dotnet/Cucumber.HtmlFormatter/MessagesToHtmlWriter.cs

```diff
@@ -32,47 +65,104 @@ public MessagesToHtmlWriter(StreamWriter writer, Action<StreamWriter, Envelope>
+        this._settings = settings ?? new HtmlReportSettings();
         _template = GetResource("index.mustache.html");
         _jsonInHtmlWriter = new JsonInHtmlWriter(writer);
         _isAsyncInitialized = true;
     }
 
     private void WritePreMessage()
     {
-        WriteTemplateBetween(_writer, _template, null, "{{css}}");
+        // Process template from beginning to CSS marker
+        string processedTemplate = ApplySettingsToTemplate(_template);
+
+        // For backward compatibility, maintain the same order of template processing
+        WriteTemplateBetween(_writer, processedTemplate, null, CSS_MARKER);
         WriteResource(_writer, "main.css");
-        WriteTemplateBetween(_writer, _template, "{{css}}", "{{messages}}");
+        WriteTemplateBetween(_writer, processedTemplate, CSS_MARKER, MESSAGES_MARKER);
     }
 
     private async Task WritePreMessageAsync()
     {
-        await WriteTemplateBetweenAsync(_writer, _template, null, "{{css}}");
+        // Process template from beginning to CSS marker
+        string processedTemplate = ApplySettingsToTemplate(_template);
+
+        // For backward compatibility, maintain the same order of template processing
+        await WriteTemplateBetweenAsync(_writer, processedTemplate, null, CSS_MARKER);
         await WriteResourceAsync(_writer, "main.css");
-        await WriteTemplateBetweenAsync(_writer, _template, "{{css}}", "{{messages}}");
+        await WriteTemplateBetweenAsync(_writer, processedTemplate, CSS_MARKER, MESSAGES_MARKER);
     }
 
     private void WritePostMessage()
     {
-        WriteTemplateBetween(_writer, _template, "{{messages}}", "{{script}}");
+        // Process template from messages to end
+        string processedTemplate = ApplySettingsToTemplate(_template);
+
+        // For backward compatibility, maintain the same order of template processing
+        WriteTemplateBetween(_writer, processedTemplate, MESSAGES_MARKER, SCRIPT_MARKER);
         WriteResource(_writer, "main.js");
-        WriteTemplateBetween(_writer, _template, "{{script}}", null);
+        WriteTemplateBetween(_writer, processedTemplate, SCRIPT_MARKER, null);
     }
 
     private async Task WritePostMessageAsync()
     {
-        await WriteTemplateBetweenAsync(_writer, _template, "{{messages}}", "{{script}}");
+        // Process template from messages to end
+        string processedTemplate = ApplySettingsToTemplate(_template);
+
+        // For backward compatibility, maintain the same order of template processing
+        await WriteTemplateBetweenAsync(_writer, processedTemplate, MESSAGES_MARKER, SCRIPT_MARKER);
         await WriteResourceAsync(_writer, "main.js");
-        await WriteTemplateBetweenAsync(_writer, _template, "{{script}}", null);
+        await WriteTemplateBetweenAsync(_writer, processedTemplate, SCRIPT_MARKER, null);
+    }
+
+    private string ApplySettingsToTemplate(string template)
```

Comment:
I think this method can be replaced with a sequence of `WriteTemplateBetweenAsync` calls. 

Either:
 * The template is effectively a constant, so we don't have to check if it a template variable should be replaced.

Or:
 * The template is not a constant, then the repeated `.Replace` calls are prone to accidental injection. For example the title is set to `{{script}}`.

So I think it would be better to consider the template to be constant. The alternative leads to problems that can only be solved by using a mustache template engine which we are trying to avoid to keep the dependencies small.